### PR TITLE
prost-build: Support generating single include file

### LINKED ISSUE #395: Support simpler generate of large dependency trees
This has outstanding PRs to implement in https://github.com/danburkert/prost/pull/123 and https://github.com/danburkert/prost/pull/333, but since these seem to have not gotten feedback from maintainers for a while I thought it made sense to open an issue to track the use case, requirements, current status, etc.

Basically, today its quite challenging to handle complex dependencies. For my example use case, I am trying to build a project that has 500 `.proto`s with many inter-dependencies. Currently, there does not seem to be an easy way to do this.

It would be great if either, or both, of (1) some improvements to the codegen to simplify this use case, and (2) documentation on how to handle these large dependency trees.

### PR DESCRIPTION
**Pull request by SkamDart**
Rebase of #123 and addresses [this issue](https://github.com/danburkert/prost/pull/333#discussion_r464741148) from abandoned #333 

Fixes #395

### CODE REVIEW

**Comment by SkamDart** [comment]
@danburkert can you take a look at this?

**Comment by LegNeato** [comment]
FWIW I would love to use this as well.

**Comment by pastjean** [comment]
Oh that'd be sweet, @danburkert can I provide any help for you on this project, to get this in ? (or anything else)

**Comment by jhugard** [comment]
+1 : this will be very helpful for projects with 100+ proto files and almost as many package definitions.

Would be even more useful if glob support were to be added to compile_protos (#469).

**Review by LucioFranco — APPROVED** [review]

**Review by LucioFranco — CHANGES_REQUESTED** [review]
So this looks great but we should add tests before we can merge this.

File: prost-build/Cargo.toml [hunk 1]
```diff
@@ -28,3 +28,8 @@ which = { version = "4", default-features = false }
 
 [dev-dependencies]
 env_logger = { version = "0.8", default-features = false }
+
+[workspace]
+members = [
+    "tests/single_include"
```
Comment (LucioFranco) [inline prost-build/Cargo.toml:34]: This seems to be failing the ci build btw

Comment (SkamDart) [reply prost-build/Cargo.toml]: Yes, I did the bad thing. Think I undid the bad thing.

**Comment by LucioFranco** [comment]
Hmm not sure why the tests didn't run and I can't approve/run them. @SkamDart can you try to force push or put a new commit to see if that triggers it?

**Comment by SkamDart** [comment]
@LucioFranco made a trivial change and pushed.

**Review by LucioFranco — APPROVED** [review]

**Pull request merged**